### PR TITLE
[RAMSES] autodetect extra fields

### PR DESCRIPTION
**Pull request by cphyc**
This should fix https://github.com/yt-project/yt/issues/1554.

- [x] Code passes flake8 checker
- [x] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by cphyc** [comment]
It looks like the build fails because somehow the dataset is cached in the tests. Is it possible?
When I test locally, the newly written test (https://github.com/yt-project/yt/pull/1555/files#diff-65852dc95a4377ada8ea57f5b6e5053aR132-R143) passes if the other one is skipped (https://github.com/yt-project/yt/pull/1555/files#diff-65852dc95a4377ada8ea57f5b6e5053aR116-R129).

**Comment by ngoldbaum** [comment]
Yes, this is related to the dataset caching that I really want to get rid of. I believe the tests run with the configuration option that disables dataset caching turned on. You probably want to do that yourself locally to reproduce the failure.

To do so, in your `ytrc` file, add `skip_dataset_cache = True`.

Ultimately I'd like to get rid of dataset caching entirely but that will take a little bit of effort to make sure data object pickling continues to work.

**Comment by cphyc** [comment]
This is indeed linked to the data being cached, so how can we prevent it from being cached for this particular test? The issue being that the two checks conflict one with each other (one is checking automatic detection, the other is checking that it is possible to feed manually the fields) ?

**Comment by ngoldbaum** [comment]
It shouldn't be cached at all on the test box (although I haven't actually checked if that is the case for this issue, it's possible I'm misremembering). I haven't looked closely at this but can do so if you want a hand.

If you want to understand how the dataset caching works more deeply, see the logic around here in `static_output.py`, https://github.com/yt-project/yt/blob/master/yt/data_objects/static_output.py#L255.

**Comment by cphyc** [comment]
As for this PR, how should I manage to get the tests passed?

**Comment by ngoldbaum** [comment]
I'll try to reproduce these test failures locally.

**Comment by ngoldbaum** [comment]
@yt-fido test this please

**Comment by ngoldbaum** [comment]
@yt-fido test this please.

File: yt/frontends/ramses/tests/test_outputs.py [hunk 1]
```diff
@@ -124,9 +124,23 @@ def test_extra_fields():
...
+    families = dd[('all', 'particle_family')]
     assert all(families == 100)
 
+
+@requires_file(ramsesExtraFieldsSmall)
+def test_extra_fields_2():
+    extra_fields = ['particle_extra_field_%s' % (i + 1) for i in range(2)]
+    ds = yt.load(os.path.join(ramsesExtraFieldsSmall, 'info_00001.txt'))
+
+    print(ds.field_list)
```
[the remark this answers is not in the document]
Comment (jzuhone) [reply yt/frontends/ramses/tests/test_outputs.py]: I'll approve if we remove this print statement.

**Comment by cphyc** [comment]
@ngoldbaum @jzuhone The print statements have been removed. Do you approve?

**Pull request merged**